auth-foundation/src/main/java/com/okta/authfoundation/credential/storage/TokenEntity.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2024-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.authfoundation.credential.storage
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import androidx.room.TypeConverters
+import com.okta.authfoundation.credential.storage.typeconverters.JsonObjectTypeConverter
+import com.okta.authfoundation.credential.storage.typeconverters.StringStringMapTypeConverter
+import kotlinx.serialization.json.JsonObject
+
+@Entity
+@TypeConverters(
+    StringStringMapTypeConverter::class,
+    JsonObjectTypeConverter::class
+)
+internal data class TokenEntity(
+    @PrimaryKey
+    val id: String,
+    val encryptedToken: ByteArray,
+    val tags: Map<String, String>,
+    val payloadData: JsonObject,
+    val keyAlias: String,
+    val userAuthenticationRequired: Boolean,
+    val encryptionAlgorithm: String,
+    val isDefault: Boolean,
```

Comment:
Is there a circumstance where multiple entries can be marked as `isDefault == true`?

Reply:
No there isn't. I put isDefault as part of stored db entry for my own convenience. [The followup PR that creates RoomTokenStorage](https://github.com/okta/okta-mobile-kotlin/pull/275) handles the logic for setting, or switching defaults atomically in a single transaction in RoomTokenStorage.replace and RoomTokenStorage.add. I have added unit tests for ensuring this scenario never happens as well.